src/classes/Obfuscator.js

```diff
@@ -57,19 +62,23 @@ export default class Obfuscator {
     let isUpdated = false;
     const records = await this.sequelize.query(`SELECT * FROM ${tableName}`, {type: Sequelize.QueryTypes.SELECT});
     for (const record of records) {
-
       for (const [columnName, columnDetails] of Object.entries(columns)) {
 
         const columnRule = tableRule?.columns?.find(col => col.name === columnName);
         const columnTypeCategory = this.getTypeCategory(columnDetails.type);
         const generalRule = this.rules.general.find(rule => rule.type === columnTypeCategory);
 
         const ruleToApply = columnRule || generalRule;
-        const {obfuscationRule, ignorePattern, ignore = false} = ruleToApply || {};
+        let {obfuscationRule, ignorePattern, ignore = false} = ruleToApply || {};
+        // Remove inner whitespace from ignore pattern
+        ignorePattern = ignorePattern?.replace(/\s/g, '');
 
-        logger.debug(`Processing Table ${tableName} for value ${record[columnName]}`);
-        if (ruleToApply && this.shouldObfuscate(tableName, columnName, record, ignorePattern)) {
+        logger.debug(`Processing Table ${tableName}[${columnName}] for value ${record[columnName]}`);
+        if (ruleToApply && this.shouldObfuscate(tableName, columnName, record, ignore, ignorePattern)) {
           logger.debug(`Applied rule ${obfuscationRule}, Pattern to ignore ${ignorePattern} for value ${record[columnName]}`);
+          if (ObfuscatorStrategyMap[obfuscationRule] === undefined) {
+            throw new Error(`Invalid obfuscation rule specified: ${obfuscationRule}`);
+          }
           record[columnName] = ObfuscatorStrategyMap[obfuscationRule].obfuscateString(record[columnName]);
```

Comment:
This will always use string obfuscator, which is not right we have them configured in the strategy.

Reply:
No. They are all in `ObfuscatorStrategyMap.js`. I don't understand this comment. All obfuscators are for strings. we don't obfuscate numbers or anything (yet).